shpkpr/vault.py

```diff
@@ -14,7 +19,10 @@ def resolve_secrets(vault_client, rendered_template):
         # parse the secret source and retrieve from vault
         path, key = definition["source"].split(":")
         path = "secret/{0}".format(path)
-        secret = vault_client.read(path)["data"][key]
-        resolved_secrets[name] = secret
+        secret = vault_client.read(path)
+        if secret:
+            resolved_secrets[name] = secret["data"][key]
+        else:
+            logger.info(f"Couldn't locate secret in Vault: {path}")
```

Comment:
Just to clarify; Will this get logged to stdout of jenkins?

Reply:
Gonna need to use old-style string formatting here to support pre3.6 i think?

https://github.com/shopkeep/shpkpr/blob/master/.travis.yml#L7

Reply:
ah, good point. will swap the formatting

Reply:
@AlanFoster i believe so, yes

Reply:
Nice, was going to ask the same question